pytorch_lightning/accelerators/accelerator.py

```diff
@@ -99,30 +100,38 @@ def backward(self, closure_loss, optimizer, opt_idx, *args, **kwargs):
         return closure_loss
 
     def optimizer_step(self, optimizer, batch_idx, opt_idx, lambda_closure, *args, **kwargs):
-        model_ref = self.trainer.get_model()
-        is_lbfgs = isinstance(optimizer, torch.optim.LBFGS)
-        using_native_amp = self.trainer.amp_backend == AMPType.NATIVE
+
         automatic_optimization = self.trainer.train_loop.automatic_optimization
+        using_native_amp = self.trainer.amp_backend == AMPType.NATIVE
 
-        # native amp + lbfgs is a no go right now
-        if using_native_amp and is_lbfgs:
-            raise MisconfigurationException(
-                'native PyTorch amp and lbfgs are not compatible.'
-                ' To request, please file a Github issue in PyTorch and tag @mcarilli')
-
-        # model hook
-        model_ref.optimizer_step(
-            epoch=self.trainer.current_epoch,
-            batch_idx=batch_idx,
-            optimizer=optimizer,
-            optimizer_idx=opt_idx,
-            optimizer_closure=lambda_closure,
-            on_tpu=False,  # TPUAccelerator class sets this as True
-            using_native_amp=using_native_amp,
-            using_lbfgs=is_lbfgs,
-            *args,
-            **kwargs,
-        )
+        if self.ddp_plugin is not None and self.ddp_plugin.using_rpc:
+            on_tpu = False
+            self.ddp_plugin.optimizer_step(optimizer, batch_idx, opt_idx, lambda_closure, on_tpu, *args, **kwargs)
```

Comment:
Shall we be explicit with our variables like below?

```python
self.ddp_plugin.optimizer_step(
    optimizer=optimizer,
    batch_idx=batch_idx,
    ...
)
```

I recall running into bugs because the variables were being added as arguments incorrectly with the below function, so I'd suggest the explicitness!